### DISCUSSION TITLE
Web3Modal session expires on IOS/Android

### DISCUSSION THREAD

**Discussion by ShahzaibAyyub** [WalletConnect/walletconnect-monorepo#2499] [Developer Support]
Upon establishing a connection with MetaMask through Web3Modal, everything operates smoothly for the first active session. However, if I revisit my web application after 1-2 days (tab remains open in safari/chrome browser), I encountered an issue where the app indicates that the wallet is connected, as it displays the wallet address and balances. Yet, if I attempt to execute my contract write functions - which were previously working perfectly - it redirects me to the MetaMask app, but no pop-up appears to seek transaction consent. When I return to my web application on the mobile browser from which I was redirected, it indicates that the wallet is not connected. Could this issue be related to session expiration after a certain duration (As this is very random, sometimes works fine for weeks)? Some users report that this problem arises even after only 1-2 hours. If that's the case, how can I adjust the session timer to ensure it remains indefinitely active, akin to its functionality on a desktop browser, where this issue does not occur?

Occurs on all mobile devices (IOS/Android) using chrome/safari/firefox for connection to metamask app. 

Here are the package versions used:
```
    "@web3modal/ethereum": "2.2.1",
    "@web3modal/react": "2.2.1",
    "wagmi": "^0.12.12",
```

**Comment by guiguy2018** [comment]
similar issue using mobile chrome on ios iphone.  Even with the web3modal test/demo site, I can't get metamask to connect reliably.  Metamask launches but does not return with a connection.
